core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -461,6 +488,7 @@ object SparkSubmit extends CommandLineUtils {
       OptionAssigner(args.queue, YARN, ALL_DEPLOY_MODES, sysProp = "spark.yarn.queue"),
       OptionAssigner(args.numExecutors, YARN, ALL_DEPLOY_MODES,
         sysProp = "spark.executor.instances"),
+      OptionAssigner(args.pyFiles, YARN, ALL_DEPLOY_MODES, sysProp = "spark.yarn.dist.pyFiles"),
```

Comment:
"spark.submit.pyFiles" both used in python and YARN#client, where in python it requires local files, but in YARN#client we can support the remote files, so using one configuration to control two scenarios is hard to do. In YARN I use a new configuration only for pyfiles to add to distributed cache.